Added rowsAffected to PostgreSQLResult

arturaz opened the pull request:
Currently `PostgreSQLExecutionContext.query` does not have a way to know how many rows did that particular
query affected. The only way to get that information is to use `PostgreSQLExecutionContext.execute`.

Unfortunately `execute` uses slower, less efficient encoding to talk to the PostgreSQL
server and does not support all data types that `query` supports.

This allows you to use `PostgreSQLExecutionContext.query` for INSERT, UPDATE, DELETE statements
and get the affected row count.

arturaz committed:
Added rowsAffected to PostgreSQLResult which is returned by `PostgreSQLExecutionContext.query`


Currently `PostgreSQLExecutionContext.query` does not have a way to know how many rows did that particular
query affected. The only way to get that information is to use `PostgreSQLExecutionContext.execute`.

Unfortunately `execute` uses slower, less efficient encoding to talk to the PostgreSQL
server and does not support all data types that `query` supports.

This allows you to use `PostgreSQLExecutionContext.query` for INSERT, UPDATE, DELETE statements
and get the affected row count.

isoos commented on lib/src/execution_context.dart:
```diff
@@ -115,5 +115,7 @@ abstract class PostgreSQLResultRow implements List {
 /// Rows can be accessed through the `[]` operator.
 abstract class PostgreSQLResult implements List<PostgreSQLResultRow> {
+  /// How many rows did this query affect?
+  int get rowsAffected;
```
I think `affectedRowCount` is a more precise name for this, e.g. we have `onlyReturnAffectedRowCount`, this aligns with that better.

isoos commented on lib/src/connection.dart:
```diff
@@ -535,9 +536,10 @@ class _PostgreSQLResultMetaData {
 
 class _PostgreSQLResult extends UnmodifiableListView<PostgreSQLResultRow>
     implements PostgreSQLResult {
+  final int rowsAffected;
```
Annotate with `@override`

isoos reviewed (commented):
Looks good, I'm asking for a few small changes, and please also use `dartfmt` to format the source code.

arturaz committed:
Renamed rowsAffected -> affectedRowCount

Asked in https://github.com/stablekernel/postgresql-dart/pull/143#discussion_r483989661

arturaz committed:
dartfmt

arturaz commented:
> Looks good, I'm asking for a few small changes, and please also use `dartfmt` to format the source code.

Updated.

isoos reviewed (approved):
Thanks, this looks good!

isoos commented:
Published as part of 2.2.0
